datafusion/expr/src/function.rs

```diff
@@ -23,6 +23,14 @@ use arrow::datatypes::{DataType, Field, Schema};
 use datafusion_common::Result;
 use std::sync::Arc;
 
+#[derive(Debug, Clone, Copy)]
```

Comment:
Eventually it would be great to add some more comments here, but maybe as a follow on PR